javaparser-core/src/main/java/com/github/javaparser/Position.java

```diff
@@ -134,31 +135,26 @@ public boolean invalid() {
     /**
      * @return If this position is valid, this.
      *   Otherwise, if the alternativePosition is valid, return that.
-     *   Otherwise otherwise, just return this.
-     *   TODO: Simplify/clarify.
+     *   Otherwise, just return this.
      */
     public Position orIfInvalid(Position alternativePosition) {
         assertNotNull(alternativePosition);
-        // TODO: Why the || ?
-        //  ... It seems that if both this and the alternative are invalid, then we return this..?
-        if (valid() || alternativePosition.invalid()) {
+        if (this.valid()) {
             return this;
         }
-        return alternativePosition;
+        return alternativePosition.valid() ? alternativePosition : this;
```

Comment:
@MysterAitch comments "orIfInvalid() -- when both this and the "other" position are both invalid, it returns "this" position.. is this intentional? My interpretation is that it should return the "other" position, even if that is invalid.". Here the method returns always this if the alternative position is not valid. Is it your intention? Can you add the note in comment.